pkg/controller/direct/compute/firewallpolicyrule_controller.go

```diff
@@ -215,6 +217,10 @@ func (a *firewallPolicyRuleAdapter) Update(ctx context.Context, updateOp *direct
 			return fmt.Errorf("error convert priority %s of ComputeFirewallPolicyRule %s to an integer: %w", tokens[5], a.id, err)
 		}
 
+		// The field priority should be removed from the patch request body and included as a query parameter.
```

Comment:
Re-clearing `Priority` here is correct. It ensures that although we used the priority for the diff check, we still comply with the API requirement to pass it as a query parameter (and not in the body) for the PATCH request.